Unknown swithc

`--no-self-contained` and `--self-contained` are considered as unknown flags when used with `dotnet build` . Those seems to be available for `dotnet publish`. Please fix the docs

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 910cd365-d63a-00dc-417c-db017f83bf6f
* Version Independent ID: 5e495ddd-140e-de5c-a793-d4a9f88be51e
* Content: [dotnet build command - .NET CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build)
* Content Source: [docs/core/tools/dotnet-build.md](https://github.com/dotnet/docs/blob/main/docs/core/tools/dotnet-build.md)
* Product: **dotnet-fundamentals**
* Technology: **dotnet-cli**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

The Web.csporj is .NET5. Here is the output I have. 
> dotnet build Web.csproj -r linux-x64 --no-self-contained
> Microsoft (R) Build Engine version 16.11.0+0538acc04 for .NET
> Copyright (C) Microsoft Corporation. All rights reserved.
> 
> MSBUILD : error MSB1001: Unknown switch.
> Switch: --no-self-contained
> 
> For switch syntax, type "MSBuild -help"

and if I do `--self-contained`:

> dotnet build Web.csproj -r linux-x64 --self-contained false
> Microsoft (R) Build Engine version 16.11.0+0538acc04 for .NET
> Copyright (C) Microsoft Corporation. All rights reserved.
> 
> MSBUILD : error MSB1001: Unknown switch.
> Switch: --self-contained
> 
> For switch syntax, type "MSBuild -help"

This is my `dotnet --info` 

>  dotnet --info
> .NET SDK (reflecting any global.json):
>  Version:   5.0.401
>  Commit:    4bef5f3dbf
> 
> Runtime Environment:
>  OS Name:     Windows
>  OS Version:  10.0.19043
>  OS Platform: Windows
>  RID:         win10-x64
>  Base Path:   C:\Program Files\dotnet\sdk\5.0.401\
> 
> Host (useful for support):
>   Version: 5.0.10
>   Commit:  e1825b4928
> 
> .NET SDKs installed:
>   2.1.526 [C:\Program Files\dotnet\sdk]
>   3.1.413 [C:\Program Files\dotnet\sdk]
>   5.0.100 [C:\Program Files\dotnet\sdk]
>   5.0.104 [C:\Program Files\dotnet\sdk]
>   5.0.207 [C:\Program Files\dotnet\sdk]
>   5.0.303 [C:\Program Files\dotnet\sdk]
>   5.0.401 [C:\Program Files\dotnet\sdk]
> 
> .NET runtimes installed:
>   Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
>   Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.WindowsDesktop.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
> 
> To install additional .NET runtimes or SDKs:
>   https://aka.ms/dotnet-download

--- a/docs/core/tools/dotnet-build.md
+++ b/docs/core/tools/dotnet-build.md
@@ -114,7 +114,7 @@ The project or solution file to build. If a project or solution file isn't speci
 
 - **`--no-self-contained`**
 
-  Publishes the application as a framework dependent application. A compatible .NET runtime must be installed on the target machine to run the application.
+  Publishes the application as a framework dependent application. A compatible .NET runtime must be installed on the target machine to run the application. Available since .NET 6.0 SDK.
 
 - **`-o|--output <OUTPUT_DIRECTORY>`**
 
@@ -124,11 +124,11 @@ The project or solution file to build. If a project or solution file isn't speci
 
 - **`-r|--runtime <RUNTIME_IDENTIFIER>`**
 
-  Specifies the target runtime. For a list of Runtime Identifiers (RIDs), see the [RID catalog](../rid-catalog.md). If you use this option, use `--self-contained` or `--no-self-contained` also.
+  Specifies the target runtime. For a list of Runtime Identifiers (RIDs), see the [RID catalog](../rid-catalog.md). If you use this option with .NET 6.0 SDK, use `--self-contained` or `--no-self-contained` also.
 
 - **`--self-contained [true|false]`**
 
-  Publishes the .NET runtime with the application so the runtime doesn't need to be installed on the target machine. The default is `true` if a runtime identifier is specified.
+  Publishes the .NET runtime with the application so the runtime doesn't need to be installed on the target machine. The default is `true` if a runtime identifier is specified.  Available since .NET 6.0 SDK.
 
 - **`--source <SOURCE>`**
 

